tests/src/autoload/functions.php

```diff
@@ -776,11 +776,12 @@ function loadDataset()
 
 /**
  * Test helper, search an item from its type and name
- *
- * @param string    $type
- * @param string    $name
- * @param bool      $onlyid
- * @return CommonDBTM|int the item, or its id
+ * @template T of CommonDBTM
+ * @param class-string<T>   $type
+ * @param string            $name
+ * @param bool              $onlyid
+ * @phpstan-return ($onlyid is true ? Int : T)
+ *      Item of $type class, or its id
```

Comment:
Note: rather that this kind of solution, it would be cleaner to split the code into two dedicated methods (`getItemByTypeName` and `getItemIdByTypeName`), thus avoiding the "flag argument" smell (which we are often guilty of in GLPI).

However it obviously require more work and don't need to be done here (I'm just saying).

Reply:
This is pretty much the same remark we had IRL yesterday with @SebSept .

Too much work to be done now, and should probably be validated before being implemented, as it would requires everyone to adapt its way to write tests.